src/firebolt_ingest/table_model.py

```diff
@@ -219,6 +220,20 @@ def partition_columns(cls, values: dict) -> dict:
                     )
         return values
 
+    @root_validator
+    def sync_mode_validator(cls, values: dict) -> dict:
+        """
+        Check whether sync_mode has one of allowed values:
+            {"overwrite", "incremental"}
+        """
+        if values.get("sync_mode"):
+            values["sync_mode"] = values["sync_mode"].lower()
+
+            if values.get("sync_mode") not in {"overwrite", "incremental"}:
+                raise ValueError(f"Unknown sync mode {values.get('sync_mode')}")
```

Comment:
Let's have `overwrite` and `append` here instead. This way we can support different incremental strategies, while also keeping it in line with the cli, where we also have this check https://github.com/firebolt-db/firebolt-cli/blob/main/src/firebolt_cli/ingest.py#L84-L86